### PR TITLE
ar71xx: Move to 4.14 kernel

### PR DESCRIPTION
**Pull request by jollaman999**
Why not for ar71xx?
Please check if 'add mdio register board info' patch is not affected to other targets.

- Tested with TP-Link Archer C7 v2

Re-worked for 4.14.48.

Also tested with some other TP-Link devices with previous pull by @pkgadd .
https://github.com/openwrt/openwrt/pull/1012

Thanks.

### CODE REVIEW

**Comment by pkgadd** [comment]
Just for next time, there wouldn't have been a need for a new pull request (nor closing the older one), you could have just force-pushed your rebased branch to update the existing pull request.

File: target/linux/ar71xx/files-4.14/drivers/mtd/tplinkpart.c [hunk 1]
```diff
@@ -154,7 +154,6 @@ static int tplink_parse_partitions_offset(struct mtd_info *master,
 	parts[0].name = "u-boot";
 	parts[0].offset = 0;
 	parts[0].size = offset;
-	parts[0].mask_flags = MTD_WRITEABLE;
```
Comment (juppin) [inline target/linux/ar71xx/files-4.14/drivers/mtd/tplinkpart.c]: Why you make the bootloader partition writable?

File: target/linux/ar71xx/files-4.14/arch/mips/ath79/mach-om5p.c [hunk 2]
```diff
@@ -160,7 +160,8 @@ static struct at803x_platform_data om5p_an_at803x_data = {
 static struct mdio_board_info om5p_an_mdio0_info[] = {
 	{
 		.bus_id = "ag71xx-mdio.0",
-		.phy_addr = 7,
+		.modalias = " om5p_an",
```
Comment (juppin) [inline target/linux/ar71xx/files-4.14/arch/mips/ath79/mach-om5p.c]: `" om5p_an"` typo?

**Comment by juppin** [comment]
First, tanks for this great work.

Probably this would be never merged...
There was a discussion on [[LEDE-DEV] kernel version status](https://lists.openwrt.org/pipermail/openwrt-devel/2018-February/011312.html)

**Comment by hzdrro** [comment]
For NAND subtarget, _linux/mtd/nand.h_ has been renamed as _linux/mtd/rawmtd.h_ in 4.14 kernel.

**Comment by jollaman999** [comment]
@juppin Thanks. I hope someone who want 4.14, try this. (Sorry for my english.)

**Comment by jollaman999** [comment]
@hzdrro Thanks. I'll check.

**Comment by juppin** [comment]
@jollaman999 I will definitely test this on all my ar71xx devices :-)
Probably ar71xx will be left on 4.9 to motivate everyone to port the devices to ath79 target...

**Comment by juppin** [comment]
> ath79 has way to many issues to allow for a smooth port.

I agree that the code base has to be improved

> So it would not be a bad idea to push ar71xx to 4.14 and leave it there.

Here i´m totally with you, because ath79 will for sure not included in 18.06 release and flow offloading is a huge win for this low powered devices...

**Comment by jow-** [comment]
4.14 for ar71xx will not be included either in 18.06, it is far too late for that.

**Comment by robimarko** [comment]
I think we all know its too late for 18.06 but snapshots are really stable

**Comment by jow-** [comment]
IMHO the debugging effort to get 4.14 stable again on ar71xx should be better spent on ath79 instead, which will also help to combat the kernel size increase a little bit.

**Comment by w01230** [comment]
Good news. Please keep updating.

**Comment by jollaman999** [comment]
I want to port to ath97 too. But I din't know about device tree. I will have to study the device tree. :)

**Comment by jollaman999** [comment]
@w01230 Thanks. I will keep updating.

**Comment by adoal** [comment]
Great work! Thank you.

**Comment by jollaman999** [comment]
Fixed the lines that @juppin said.
Also thanks for the force update tip! @pkgadd

**Comment by lucize** [comment]
@jollaman999 the overall part for ath79 is ready, it seems to me that only shared irq driver has major problems (scenario with dual wifi interfaces) if you have the energy and knowledge try debugging that. there are also some quirks related to some wifi adapters

**Comment by pepe2k** [comment]
@jollaman999, ar71xx target is going to stay at 4.9 in the upcoming release, we don't plan bringing it to 4.14 in future as it will be swapped by ath79. All the efforts should be now focused on new target rather than old one which at some time will be dropped.

**Comment by jack338c** [comment]
@jollaman999 
hi,it can't be applyed into 941n v7,Could you tell how to deal with this? thanks
https://github.com/coolsnowwolf/lede/commit/698ab19b771dbfa36f4da6f90fff5d084ce8e889

**Comment by blogic** [comment]
I am against adding this to trunk. people should start using ath79 and invest time into moving their boards over for the next stable.

**Comment by bladeoner** [comment]
@jollaman999 does your pull request also enable offloading by default?

Thanks in advance.

**Comment by juppin** [comment]
@bladeoner this is enabled on 4.14 by default, no need to enable that for a target.

**Comment by bladeoner** [comment]
@juppin thanks for the quick response.

**Comment by bladeoner** [comment]
@hzdrro 
For NAND subtarget, linux/mtd/nand.h has been renamed as linux/mtd/rawnand.h in 4.14 kernel.

How did you fix the above, which files did you modify?

**Comment by hzdrro** [comment]
@bladeoner I modified all the mach files of the nand subtarget and target/linux/ar71xx/files/drivers/mtd/nand/ar934x_nfc.c by by replacing "#include <linux/mtd/nand.h>" with "#include <linux/mtd/rawnand.h>".

**Comment by lucize** [comment]
> Did someone test this PR on archer c7 v2? I met problem 2.4G and 5G can not work.

@w01230  why not try this working ath79 tree  https://github.com/SibrenVasse/openwrt/tree/ath79-archer-c7

**Comment by w01230** [comment]
@lucize I noticed this ath79 tree for archer. but I do not know if anything works well?

**Comment by lucize** [comment]
Everything is going well!

**Comment by bladeoner** [comment]
@hzdrro Thank you for your response I found the files and modified them.
I'm only running into some issues finishing the build.

**Comment by bladeoner** [comment]
It seems I fixed it but I'm having a strange issue, after setting a password in LuCI and rebooting the router I cannot login in LuCI anymore. I can login through SSH, any idea anyone?

**Comment by juppin** [comment]
If you use chrome there was an issue with luci...
Try another browser...

**Comment by bladeoner** [comment]
I’m using Firefox, I also tried Chrome same result.

**Comment by jollaman999** [comment]
Closed thanks for the ath79 port!

**Comment by lucize** [comment]
@jollaman999 thank you too ! for the moment I'm using this on 1043nd-v1 because of some switch bug

**Pull request closed without merging**